# Notebook of Tasks for Module Applied Statistics as part of Higher Diploma in Computer Science and Data Analytics

## Task 1

### Problem 1

Using the Lady tasting Tea experiment, calculate the probability of the lady being able to select all cups of tea where milk was put in the cup before the tea. There are 12 cups in total. 6 of which will have been made with milk being used first and the other 6 with tea first. 

In [1]:
# Import necessary modules

import math
import random
import itertools
import numpy as np
import matplotlib.pyplot as plt

In [2]:
# Variables

total_cups = 12

#Milk first cups
number_milk_first = 6

#Tea first cups
number_tea_first = 6

In [10]:
# Total number of ways to select any 6 cups out of the 12. The order of the cups chosen doesnt matter

ways_of_combining_all_cups = 12*11*10*9*8*7
ways_of_combining_all_cups

665280

In [20]:
# Assuming selecting just the milk first cups is a success. Then the number of ways these could be chosen in any order is:

no_successes = 6*5*4*3*2*1
no_successes


720

In [80]:
# The total combinations of selecting 6 cups out of 12 
six_out_of_twelve = ways_of_combining_all_cups // no_successes
six_out_of_twelve



924

In [90]:
#Probabiltiy

prob_six = 1/six_out_of_twelve
prob_six

0.0010822510822510823

With no special ability to select milk-first cups, the lady would have a 1 in 924 chance or .108% probability of selecting just the 6 milk-first cups

### Problem 2

By choosing 5/6 of the cups correctly it might still show an ability to discern between the two categories of tea. So we will allow one mistake. This means we need to calculate the probability for 2 scenarios

- getting all 6 correct
- getting 5 out of 6 correct

Once we have these we can add them together.

Instead of using the above calculations we will use the function math.comb() which uses the formula:

$$ 
\frac{n!}{k!(n-k)!}
$$

We already have the probability of getting all 6 correct from Problem 1.

1/924 or .108%

Now we need to find the number of ways to select 5 correct cups out of 6

In [85]:
# total combinations for 5 out of 6 cups correct
# n = 6, k = 5
five_out_of_six = math.comb(6,5)
five_out_of_six


6

In [86]:
# total combinations for 1 out of 6
# n = 6, k = 1

one_out_of_six = math.comb(6,1)
one_out_of_six

6

In [94]:
#Probability of picking 5 out 6 cups correctly given 12 cups total

prob_five = (five_out_of_six * one_out_of_six) / six_out_of_twelve
prob_five

0.03896103896103896

The probability of picking 5 out of 6 correctly rises to 36/924 or 3.89%

The scenario where either 5/6 or 6/6 is a success is calculated by adding the two together

In [95]:
Answer = prob_six+prob_five
Answer

0.04004329004329004

If allowing for one error, the probability of selecting all 6 Milk-first cups or 5 of the Milk-first cups is 37/924 or 4%

Would you allow errors?

We now need to find the odds of choosing 4 out of 6 correct

In [96]:
# total combinations for 4 out of 6 cups correct
# n = 6, k = 4
four_out_of_six = math.comb(6,4)
four_out_of_six


15

In [100]:
# total combinations for 2 out of 6
# n = 6, k = 2

two_out_of_six = math.comb(6,2)
two_out_of_six

15

In [101]:
#Probability of picking 4 out 6 cups correctly given 12 cups total

prob_four = (four_out_of_six * two_out_of_six) / six_out_of_twelve
prob_four

0.2435064935064935

In [104]:
#Probability of picking at least 4 out 6 cups correctly
prob_total = prob_six+prob_five+prob_four
prob_total

0.28354978354978355

The odds of just getting 4 of the 6 choices correct is 24.3%. If accepting anything over 4 cups correct the probability is over 28% and is not a strong statistical indicator of any ability to discern between the two categories of tea.  